internal/provisioner/controller/gatewayclass.go

```diff
@@ -57,6 +60,15 @@ func NewGatewayClassController(mgr manager.Manager, gatewayController string) (c
 		return nil, err
 	}
 
+	// Watch ContourDeployments since they can be used as parameters for
+	// GatewayClasses.
+	if err := c.Watch(
+		&source.Kind{Type: &contour_api_v1alpha1.ContourDeployment{}},
+		handler.EnqueueRequestsFromMapFunc(r.mapContourDeploymentToGatewayClasses),
```

Comment:
for the future, will we need to implement this differently (custom handler that ignores the update event?) so that updates to `ContourDeployment` resources don't trigger a reconcile? (and updates to `GatewayClasses` do not cause a re-reconfiguration of the Contour installation for an existing `Gateway`)

Reply:
Yeah good question - I think there are a couple different ways we could go about implementing, but my plan had been to revise the logic that's in the `EnsureObject` functions, to not do full updates if the object already exists, but instead to *only* update those fields that should be updated. So things coming from GatewayClass would not trigger updates, but e.g. updates to Listener definitions would trigger updates. I think we probably need some logic like that regardless, because if you take that example of Listeners being updated, we need to ensure when we do push those updates through, that things derived from GatewayClass don't also get updated to reflect the current state of the GatewayClass.